### DISCUSSION TITLE
Download my own quizzes, assignments with answers

### DISCUSSION THREAD

**Discussion by sardnar** [ucfopen/canvasapi#643] [Q&A]
Hello,
I am a student backing up my own old courses. Every thing in a module works great from files and html content and all except for quizzies and assignments. If I loop over course.get_quizzes() and course.get_assignments(), I don't find any property to download this full webpage, only the .description which is html of the starting prompt of these quizzes/assignments. 

Is there is a snippet to help me backup my own assignments?

**Comment by bennettscience** [comment]
Quiz objects in Canvas don't have answers attached. The workaround is to get your submission for the assignment instead. You can use the `include=submission_history` query param to get the submission object. [This Canvas Community thread](https://community.canvaslms.com/t5/Canvas-Developers-Group/How-get-all-answers-from-a-quiz-using-the-API/m-p/241225) may point you in the right direction.

Since it sounds like you're a student, you might be able to get all of your assignment submissions (including quizzes) using the [submissions endpoint](https://canvas.instructure.com/doc/api/submissions.html#method.submissions_api.for_students) may be more efficient.

**Reply by sardnar** [reply]
Thank you for your insight, really helpful. But to understand your point, you mean that canvasapi does not support it and I need to scrape the API myself? I just never interacted with an API before so not familair with terminology

**Reply by bennettscience** [reply]
No, you don't need to scrape it. If you're already looping over your courses, you can do something like this:

```python
canvas = Canvas(URL, KEY)
course = canvas.get_course(12345)
submissions = course.get_multiple_submissions(include="submission_history")

# carry on...
```

[Here's the page in the canvasapi docs](https://canvasapi.readthedocs.io/en/stable/course-ref.html#canvasapi.course.Course.get_multiple_submissions) with the method. It _should_ get all of your submissions for a given course. There may be some issues with Quizzes not showing correct answers if you're not a teacher - it will return data based on your access level in the course (meaning if the instructor has not turned on the "show correct answers" option, you won't get the correct answers in the API either).

I don't use Canvas any more, so I can't test this particular function, but I've used a method similar to this in the past.
